packages/vuetify/src/components/VRating/VRating.tsx

```diff
@@ -0,0 +1,251 @@
+      const isHalf = props.halfIncrements && isHalfEvent(e)
+
+      return i + (isHalf ? 0.5 : 1)
+    }
+
+    const itemState = computed(() => range.value.map(index => {
+      const isHovering = props.hover && hoverIndex.value > -1
+
+      const isFilled = Math.floor(rating.value) > index
+      const isHovered = Math.floor(hoverIndex.value) > index
+      const isHalfFilled = !isFilled && (rating.value - index) % 1 > 0
+      const isHalfHovered = !isHovered && (hoverIndex.value - index) % 1 > 0
+
+      const isFullIcon = isHovering ? isHovered : isFilled
+      const isHalfIcon = isHovering ? isHalfHovered : isHalfFilled
+
+      const icon = isFullIcon ? props.fullIcon
+        : isHalfIcon ? props.halfIcon
+        : props.emptyIcon
+
+      const color = isFilled || isHalfFilled || isHovered ? props.color : props.bgColor
+
+      return { isFilled, isHovered, isHalfFilled, isHalfHovered, icon, color }
+    }))
+
+    const eventState = computed(() => range.value.map(index => {
+      function onMouseenter (e: MouseEvent): void {
+        hoverIndex.value = genHoverIndex(e, index)
+      }
+
+      function onMouseleave (): void {
+        hoverIndex.value = -1
+      }
+
+      function onClick (e: MouseEvent) {
+        if (props.readonly) return
+
+        // If detail === 0 then click is triggered by keyboard
+        if (e.detail === 0) {
+          const currentIndex = Math.floor(rating.value)
+          if (rating.value - 1 === index && props.clearable) {
+            rating.value = 0
+          } else if (currentIndex !== index || !props.halfIncrements) {
+            rating.value = index + (props.halfIncrements ? 0.5 : 1)
+          } else {
+            rating.value += 0.5
+          }
+        } else {
+          const newValue = genHoverIndex(e, index)
+          if (props.clearable && rating.value === newValue) {
+            rating.value = 0
+          } else {
+            rating.value = newValue
+          }
+        }
+      }
+
+      return {
+        onClick,
+        onMouseenter: props.hover ? onMouseenter : undefined,
+        onMouseleave: props.hover ? onMouseleave : undefined,
+        onMousemove: props.hover && props.halfIncrements ? onMouseenter : undefined,
+        ref: (e: any) => updateRef(e, index),
+      }
+    }))
+
+    const items = computed(() => range.value.map(index => ({
+      ariaLabel: t(props.itemAriaLabel, index + 1, range.value.length),
+      density: props.density,
+      disabled: props.disabled,
+      hasLabels: !!props.itemLabels?.length || !!slots['item-label'],
+      index,
+      label: props.itemLabels && props.itemLabels[index],
+      readonly: props.readonly,
+      ripple: props.ripple,
+      size: props.size,
+      tabindex: props.readonly ? -1 : undefined,
+      text: true,
+      value: rating.value,
+      ...itemState.value[index],
+      ...eventState.value[index],
+    })))
+
+    function updateFocus () {
+      const index = Math.floor(rating.value - 0.5)
```

Comment:
Hidden `<input type="radio">` will do this for you, and support screen readers properly. For baseline screen reader support you need at least `role="radio" ariaChecked={ rating.value === index + 1 }` on the buttons.